### ISSUE TITLE
Prisma Migrate Failure exposes database connection parameters

### ISSUE THREAD

**Issue by uptownhr** [prisma/prisma#6273]
## Bug description

Upon a permission failure in accessing the database, schema or tables, `prisma migrate` will error and expose the full database connection parameters to the logs.

## How to reproduce

1. Setup PRISMA schema and adapter (I was connecting in node to postgres)
2. Create a migration
3. Run the migration but change the db permission to deny access

## Expected behavior

A database error without the credentials and connection properties being exposed

## Prisma information
```
datasource db {
  provider = "postgresql"
  url      = env("POSTGRES_URL")
}
```

## The error
actual values replaced with xxxx
```
with self=PostgresFlavour { url: PostgresUrl { url: Url { scheme: "postgresql", username: "xxxx", password: Some("xxxx"), host: Some(Domain("xxxx")), port: Some(5432), path: "/xxxx", query: Some("schema=xxx&statement_cache_size=0"), fragment: None }, query_params: PostgresUrlQueryParams { ssl_params: SslParams { certificate_file: None, identity_file: None, identity_password: <HIDDEN>, ssl_accept_mode: AcceptInvalidCerts }, connection_limit: None, schema: "xxxx", ssl_mode: Prefer, pg_bouncer: false, host: None, socket_timeout: None, connect_timeout: Some(5s), pool_timeout: Some(10s), statement_cache_size: 0 } }, features: BitFlags<MigrationFeature>(0b0) } connection=Connection(Postgres(PostgresUrl { url: Url { scheme: "xxxx", username: "xxxx", password: Some("xxxx"), host: Some(Domain("xxxx")), port: Some(5432), path: "/xxxx", query: Some("schema=xxxx&statement_cache_size=0"), fragment: None }, query_params: PostgresUrlQueryParams { ssl_params: SslParams { certificate_file: None, identity_file: None, identity_password: <HIDDEN>, ssl_accept_mode: AcceptInvalidCerts }, connection_limit: None, schema: "bambee-payroll", ssl_mode: Prefer, pg_bouncer: false, host: None, socket_timeout: None, connect_timeout: Some(5s), pool_timeout: Some(10s), statement_cache_size: 0 } }))
```

## Environment & setup

- OS: Alpine Docker
- Database: Postgres (aurora)
- Node.js version: 14.15.1
- Prisma version: 2.19.0

```

```

**Comment by janpio** [comment]
I can confirm similar behavior for `db push`, see error message from https://github.com/prisma/prisma/issues/6277 which included the actual password instead of `...`.

**Comment by tomhoule** [comment]
This is bad — for now I will do a quick review of the migrate codebase to fix this and search for similar cases.

I also created an internal issue for us to spend time finding mechanisms to make sure it doesn't happen again.

**Comment by tomhoule** [comment]
@uptownhr https://github.com/prisma/prisma-engines/pull/1787 should fix this short term. As noted above we scheduled some time to ensure we have mechanisms in place to prevent this from happening again in the future.

**Comment by tomhoule** [comment]
The fix will be in tomorrow's release (2.20.0).
